libyul/optimiser/NameSimplifier.h

```diff
@@ -47,7 +48,10 @@ class NameSimplifier: public ASTModifier
 	static constexpr char const* name{"NameSimplifier"};
 	static void run(OptimiserStepContext& _context, Block& _ast)
 	{
-		NameSimplifier{_context, _ast}(_ast);
+		NameSimplifier nameSimplifier{_context, _ast};
+		nameSimplifier(_ast);
+		for (auto const& names: nameSimplifier.m_translations)
+			nameSimplifier.m_context.dispenser.markUsed(names.second);
```

Comment:
I think this should be part of `NameSimplifier::findSimplification` instead (and that function could be split up a little).